src/OSPSuite.Core/Domain/Builder/SimulationBuilder.cs

```diff
@@ -135,8 +207,49 @@ private void addMolecules(IMoleculeDependentBuilder builder, MoleculeList molecu
       internal IReadOnlyCollection<MoleculeBuilder> Molecules => _molecules;
       internal IReadOnlyCollection<ParameterValue> ParameterValues => _parameterValues;
       internal IReadOnlyCollection<InitialCondition> InitialConditions => _initialConditions;
+
       internal MoleculeList MoleculeListFor(IMoleculeDependentBuilder builder) => _moleculeListCache[builder];
 
       internal MoleculeBuilder MoleculeByName(string name) => _molecules[name];
+
+      internal class BuilderSource
```

Comment:
Temp class holding a builder and it's building block. I am using it to retrieve the info about bb and module when tracking